pkg/operator/controller/canary/daemonset_test.go

```diff
@@ -0,0 +1,60 @@
+package canary
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestDesiredCanaryDaemonSet(t *testing.T) {
+	canaryImage := "openshift/hello-openshift:latest"
+
+	daemonset := desiredCanaryDaemonSet(canaryImage)
+
+	expectedDaemonSetName := types.NamespacedName{
```

Comment:
Is this the same ns/name as `controller.CanaryDaemonSetName()`? If so, I suggest calling the function instead.

Reply:
fixed.